pkg/maps/metricsmap/metricsmap.go

```diff
@@ -0,0 +1,93 @@
+// Copyright 2016-2018 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package metricsmap
+
+import (
+	"fmt"
+	"unsafe"
+
+	"github.com/cilium/cilium/pkg/bpf"
+	"github.com/cilium/cilium/pkg/logging"
+)
+
+var log = logging.DefaultLogger
+
+const (
+	// MapName for metrics map.
+	MapName = "cilium_metrics"
+	// MaxEntries is the maximum number of keys that can be present in the
+	// Metrics Map.
+	MaxEntries = 65536
+)
+
+// Key must be in sync with struct metrics_key in <bpf/lib/common.h>
+type Key struct {
+	reason   uint8
+	dir      uint8
+	reserved [3]uint16
+}
+
+// Value must be in sync with struct metrics_value in <bpf/lib/common.h>
+type Value struct {
+	count uint64
+	bytes uint64
+}
+
+// String converts the key into a human readable string format
+func (k *Key) String() string {
+	return fmt.Sprintf("reason:%d dir:%d", k.reason, k.dir)
```

Comment:
I would change the returned string to be human-readable otherwise we will only see `reason:0 dir:0` without knowing the meaning of it.

Reply:
Sure..
Will this work?
```
fmt.Sprintf("Drop/Forward Reason:%d Direction(Ingress /Egress):%d", k.reason, k.dir)
```

Reply:
Also, if something more elaborate is needed, I can do it as part of #4171  which is the followup of this PR and will be implementing the user space bits of reading the map. I will start working on the user space controller which pulls these metrics and export them to prometheus.

Reply:
ok, I'm fine doing it as part of #4171

Reply:
Will do . Thanks!